core/src/main/java/hudson/util/SequentialExecutionQueue.java

```diff
@@ -107,7 +107,9 @@ private QueueEntry(Runnable item) {
         // Caller must have a lock
         private void submit() {
             submissionTime = System.currentTimeMillis();
-            executors.submit(this);
+            synchronized (SequentialExecutionQueue.this) {
+                executors.submit(this);
+            }
```

Comment:
Yes, the caller must have a lock as the comment states. And callers do have a lock. But since synchronization is re-entrant, it doesn't hurt to explicitly synchronize here, and it also makes SpotBugs happy.